File: src/main/java/org/orbeon/oxf/externalcontext/ResponseWrapper.java
/**
 * Copyright (C) 2009 Orbeon, Inc.
 *
 * This program is free software; you can redistribute it and/or modify it under the terms of the
 * GNU Lesser General Public License as published by the Free Software Foundation; either version
 * 2.1 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
 * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
 * See the GNU Lesser General Public License for more details.
 *
 * The full text of the license is available at http://www.gnu.org/copyleft/lesser.html
 */
package org.orbeon.oxf.externalcontext;

import java.io.OutputStream;
import java.io.PrintWriter;

public class ResponseWrapper implements ExternalContext.Response {
    private ExternalContext.Response _response;

    public ResponseWrapper(ExternalContext.Response response) {
        if (response == null)
            throw new IllegalArgumentException();
        this._response = response;
    }

    public boolean checkIfModifiedSince(ExternalContext.Request request, long lastModified) {
        return _response.checkIfModifiedSince(request, lastModified);
    }

    public String getCharacterEncoding() {
        return _response.getCharacterEncoding();
    }

    public String getNamespacePrefix() {
        return _response.getNamespacePrefix();
    }

    public OutputStream getOutputStream() {
        return _response.getOutputStream();
    }

    public PrintWriter getWriter() {
        return _response.getWriter();
    }

    public boolean isCommitted() {
        return _response.isCommitted();
    }

    public void reset() {
        _response.reset();
    }

    public String rewriteActionURL(String urlString) {
        return _response.rewriteActionURL(urlString);
    }

    public String rewriteRenderURL(String urlString) {
        return _response.rewriteRenderURL(urlString);
    }

    public String rewriteActionURL(String urlString, String portletMode, String windowState) {
        return _response.rewriteActionURL(urlString, portletMode, windowState);
    }

    public String rewriteRenderURL(String urlString, String portletMode, String windowState) {
        return _response.rewriteRenderURL(urlString, portletMode, windowState);
    }

    public String rewriteResourceURL(String urlString, int rewriteMode) {
        return _response.rewriteResourceURL(urlString, rewriteMode);
    }

    public void sendError(int len) {
        _response.sendError(len);
    }

    public void sendRedirect(String location, boolean isServerSide, boolean isExitPortal) {
        _response.sendRedirect(location, isServerSide, isExitPortal);
    }

    public void setPageCaching(long lastModified) {
        _response.setPageCaching(lastModified);
    }

    public void setResourceCaching(long lastModified, long expires) {
        _response.setResourceCaching(lastModified, expires);
    }

    public void setContentLength(int len) {
        _response.setContentLength(len);
    }

    public void setContentType(String contentType) {
        _response.setContentType(contentType);
    }

    public void setHeader(String name, String value) {
        _response.setHeader(name, value);
    }

    public void addHeader(String name, String value) {
        _response.addHeader(name, value);
    }

    public void setStatus(int status) {
        _response.setStatus(status);
    }

    public void setTitle(String title) {
        _response.setTitle(title);
    }

    public Object getNativeResponse() {
        return _response.getNativeResponse();
    }

    public ExternalContext.Response _getResponse() {
        return _response;
    }
}

File: common/shared/src/main/scala/org/orbeon/oxf/externalcontext/ResponseWrapper.scala
package org.orbeon.oxf.externalcontext

import java.io.{OutputStream, PrintWriter}


class ResponseWrapper(var _response: ExternalContext.Response)
  extends ExternalContext.Response {

  require(_response ne null)

  def checkIfModifiedSince(request: ExternalContext.Request, lastModified: Long): Boolean = _response.checkIfModifiedSince(request, lastModified)
  def getCharacterEncoding: String = _response.getCharacterEncoding
  def getNamespacePrefix: String = _response.getNamespacePrefix
  def getOutputStream: OutputStream = _response.getOutputStream
  def getWriter: PrintWriter = _response.getWriter
  def isCommitted: Boolean = _response.isCommitted

  def reset(): Unit =
    _response.reset()

  def rewriteActionURL(urlString: String): String = _response.rewriteActionURL(urlString)
  def rewriteRenderURL(urlString: String): String = _response.rewriteRenderURL(urlString)
  def rewriteActionURL(urlString: String, portletMode: String, windowState: String): String = _response.rewriteActionURL(urlString, portletMode, windowState)
  def rewriteRenderURL(urlString: String, portletMode: String, windowState: String): String = _response.rewriteRenderURL(urlString, portletMode, windowState)
  def rewriteResourceURL(urlString: String, rewriteMode: UrlRewriteMode): String = _response.rewriteResourceURL(urlString, rewriteMode)

  def sendError(len: Int): Unit =
    _response.sendError(len)

  def sendRedirect(location: String, isServerSide: Boolean, isExitPortal: Boolean): Unit =
    _response.sendRedirect(location, isServerSide, isExitPortal)

  def setPageCaching(lastModified: Long): Unit =
    _response.setPageCaching(lastModified)

  def setResourceCaching(lastModified: Long, expires: Long): Unit =
    _response.setResourceCaching(lastModified, expires)

  def setContentLength(len: Int): Unit =
    _response.setContentLength(len)

  def setContentType(contentType: String): Unit =
    _response.setContentType(contentType)

  def setHeader(name: String, value: String): Unit =
    _response.setHeader(name, value)

  def addHeader(name: String, value: String): Unit =
    _response.addHeader(name, value)

  def setStatus(status: Int): Unit =
    _response.setStatus(status)

  def getStatus: Int = _response.getStatus

  def setTitle(title: String): Unit =
    _response.setTitle(title)

  def getNativeResponse: AnyRef = _response.getNativeResponse
  def _getResponse: ExternalContext.Response = _response
}